CHANGELOG/v1.6.6.md

```diff
@@ -0,0 +1,42 @@
+## 👌 Kubernetes version support
+
+- Management Cluster: v1.25.x -> v1.29.x
+- Workload Cluster: v1.23.x -> v1.29.x
+
+[More information about version support can be found here](https://cluster-api.sigs.k8s.io/reference/versions.html)
+
+## Changes since v1.6.5
+## :chart_with_upwards_trend: Overview
+- 14 new commits merged
+- 5 bugs fixed 🐛
+
+## :bug: Bug Fixes
+- ClusterClass: Defaulting webhook should check class is set in ClusterClass-based clusters (#10674)
+- ClusterClass: Fix indexing in ClusterClass webhook MHC validation (#10677)
+- MachineHealthCheck: Skip publishing the `RemediationRestricted` event when there are no unhealthy target (#10617)
+- MachinePool: Check that replicas pointer is not nil in machinepool controller (#10635)
+- KCP/CAPD: KCPTemplate & DockerClusterTemplate webhook: default before immutability check (#10647)
+
+## :seedling: Others
+- clusterctl: Bump cert-manager to v1.15.0 (#10733)
+- Dependency: Bump to Go 1.21.11 (#10741)
+- Devtools: Use setup-envtest from CR main (05/2024) to use envtest binaries from CT releases (#10664)
+- Security: Cleanup owner files (#10680)
+- Testing: Support MachinePools without MachinePoolMachines in clusterctl upgrade test (#10615)
+- Testing: Use kind as a secondary management cluster for clusterctl E2E tests (#10639)
```

Comment:
```suggestion
```
This PR was merged against main. If this was generated with the release notes tool we should take a look if we have a bug

(I didn't double check all other PRs if they were actually merged against release-1.6)

Reply:
/hold

Reply:
/hold cancel
Double checked all other PRs, the other ones are from release-1.6. But to be honest I don't have the time to do this :)

Reply:
Just ran the tool locally and indeed it adds the PR from main. Let's open an issue and investigate this please

Reply:
I will check the release notes again and also keep it in mind for next releases. I will create an issue and also investigate why this is happening.

Reply:
@vishalanarase Otherwise lgtm once this line is removed from my side :)

Reply:
Updated the MR